OloEngine/src/OloEngine/Audio/SoundGraph/SoundGraphSound.cpp

```
@@ -38,6 +41,30 @@ namespace OloEngine::Audio::SoundGraph
     {
         OLO_PROFILE_FUNCTION();
         m_Source = CreateScope<Audio::SoundGraph::SoundGraphSource>();
+
+        // Hook the source's custom ma_node into miniaudio's node graph using the live
+        // AudioEngine instance. Without this, the source allocates the custom-node struct
+        // but ma_node_init never runs and the audio thread never pulls samples — i.e. the
+        // graph plays silently. We fetch the engine through the AudioEngine facade rather
+        // than threading it through every caller.
+        auto* engine = static_cast<ma_engine*>(AudioEngine::GetEngine());
+        if (!engine)
+        {
```

Cppcheck (2.20.0)

[style] The function 'GetSourceCount' is never used.

(unusedFunction)